### PR TITLE
vsphere-upi: Use openshift-install coreos print-stream-json if available

### PR DESCRIPTION
**Pull request by cgwalters**
See https://github.com/openshift/installer/blob/master/docs/user/overview.md#coreos-bootimages
and https://github.com/openshift/enhancements/blob/master/enhancements/coreos-bootimages.md

### CODE REVIEW

**Comment by cgwalters** [comment]
Cool, this worked!  I see in the logs here: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/17349/rehearse-17349-pull-ci-openshift-installer-master-e2e-vsphere-upi/1377288314939248640/artifacts/e2e-vsphere-upi/upi-conf-vsphere/build-log.txt
> 2021-03-31 16:12:59+00:00 - Creating reusable variable files...
> Using stream metadata

It just failed on unrelated e2e tests.  So, let's ship this.

/test ci/build-farm/app-ci-config-dry

**Review by staebler — COMMENTED** [review]
Looks good. Thanks for this!

File: ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh [hunk 1]
```diff
@@ -41,9 +41,24 @@ install_config="${SHARED_DIR}/install-config.yaml"
...
+if openshift-install coreos print-stream-json 2>/tmp/err.txt >${SHARED_DIR}/coreos.json; then
+   echo "Using stream metadata"
+   ova_url=$(jq -r '.architectures.x86_64.artifacts.vmware.formats.ova.disk.location' < ${SHARED_DIR}/coreos.json)
+else
+  if ! grep -qF 'unknown command "coreos"' /tmp/err.txt; then
+    echo "Unhandled error from openshift-install" 1>&2
+    cat /tmp/err.txt
+    exit 1
+  fi
+  rm -f /tmp/err.txt
```
Comment (staebler) [inline ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh:56]: Should this be deleted in all code paths?

**Review by staebler — COMMENTED** [review]
Apparently, I did not save my comment.

File: ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh (hunk 1 above)
Comment (cgwalters) [reply ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh]: Yep, let me remove that error...

**Review by staebler — APPROVED** [review]
/lgtm

**Pull request merged**